app/src/main/java/fr/neamar/kiss/utils/FuzzyScore.java

```diff
@@ -0,0 +1,291 @@
+		 * Can only compare scores with same search pattern.
+		 */
+		public int    score;
+		//		/**
+//		 * input str with matched characters marked between {}
+//		 */
+//		public String formattedString;
+		public ArrayList<Integer> matchedIndices;
+
+		public List<Pair<Integer, Integer>> getMatchedSequences()
+		{
+			// compute pair match indices
+			List<Pair<Integer, Integer>> positions = new ArrayList<>( matchedIndices.size() );
+			int                          start     = this.matchedIndices.get( 0 );
+			int                          end       = start + 1;
+			for( int i = 1; i < this.matchedIndices.size(); i += 1 )
+			{
+				if( end == this.matchedIndices.get( i ) )
+				{
+					end += 1;
+				}
+				else
+				{
+					positions.add( new Pair<>( start, end ) );
+					start = this.matchedIndices.get( i );
+					end = start + 1;
+				}
+			}
+			positions.add( new Pair<>( start, end ) );
+			return positions;
+		}
+	}
+
+	/**
+	 * Returns true if each character in pattern is found sequentially within str
+	 *
+	 * @param pattern
+	 * @param str
+	 * @return true if each character in pattern is found sequentially within str
+	 */
+	public boolean match( String pattern, String str )
+	{
+		int patternIdx    = 0;
+		int strIdx        = 0;
+		int patternLength = pattern.length();
+		int strLength     = str.length();
+
+		while( patternIdx != patternLength && strIdx != strLength )
+		{
+			int patternChar = Character.toLowerCase( pattern.codePointAt( patternIdx ) );
+			int strChar     = Character.toLowerCase( str.codePointAt( strIdx ) );
+			if( patternChar == strChar )
+				++patternIdx;
+			++strIdx;
+		}
+
+		return patternLength != 0 && strLength != 0 && patternIdx == patternLength;
+	}
+
+	/**
+	 * @param pattern
+	 * @param text
+	 * @return true if each character in pattern is found sequentially within text
+	 */
+	public boolean match( String pattern, String text, MatchInfo info )
+	{
+		if( info == null )
+			return match( pattern, text );
+
+		// Loop variables
+		int     score         = 0;
+		int     patternIdx    = 0;
+		int     patternLength = pattern.length();
+		int     strIdx        = 0;
+		int     strLength     = text.length();
+		boolean prevMatched   = false;
+		boolean prevLower     = false;
+		boolean prevSeparator = true;       // true so if first letter match gets separator bonus
+
+		// Use "best" matched letter if multiple string letters match the pattern
+		Integer bestLetter      = null;
+		Integer bestLower       = null;
+		Integer bestLetterIdx   = null;
+		int     bestLetterScore = 0;
+
+		ArrayList<Integer> matchedIndices = new ArrayList<>( pattern.length() );
+
+		// Loop over strings
+		while( strIdx != strLength )
+		{
+			Integer patternChar = patternIdx != patternLength ? pattern.codePointAt( patternIdx ) : null;
+			int     strChar     = text.codePointAt( strIdx );
+
+			Integer patternLower = patternChar != null ? Character.toLowerCase( patternChar ) : null;

```

Comment:
See comment above for building patternChar and patternLower once and for all in the constructor